tests/cmd/auth/test_login.py

```diff
@@ -175,58 +171,68 @@ def test_auth_login_token_update_existing_config(self, monkeypatch, cli_fs_runne
 
 
 class TestAuthLoginWeb:
-    def test_auth_login_web_not_enabled(self, cli_fs_runner):
-        """
-        GIVEN -
-        WHEN the ggshield web login feature flag is off
-        THEN it is not possible to login using the web method
-        """
-        cmd = ["auth", "login", "--method=web"]
-        result = cli_fs_runner.invoke(cli, cmd, color=False, catch_exceptions=True)
-        assert result.exit_code == 1
-        assert result.output == "Error: The web auth login method is not enabled.\n"
-
     @pytest.mark.parametrize(
-        ["port_used_count", "authorization_code", "is_exchange_ok", "is_token_valid"],
         [
-            (0, "some_valid_authorization_code", True, True),
-            (1, "some_valid_authorization_code", True, True),
-            (1000, "some_valid_authorization_code", True, True),
-            (0, "some_valid_authorization_code", False, False),
-            (0, "some_valid_authorization_code", True, False),
-            (0, None, True, True),  # invalid authorization code from callback
+            "token_name",
+            "port_used_count",
+            "authorization_code",
+            "is_state_valid",
+            "is_exchange_ok",
+            "is_token_valid",
+        ],
+        [
+            # valid case
+            ("test token", 0, "some_valid_authorization_code", True, True, True),
+            # default token name
+            (None, 0, "some_valid_authorization_code", True, True, True),
+            # first port already used
+            ("test token", 1, "some_valid_authorization_code", True, True, True),
+            # all ports already used
+            ("test token", 1000, "some_valid_authorization_code", True, True, True),
+            # can't get token from server
+            ("test token", 0, "some_valid_authorization_code", True, False, False),
+            # invalid token from server
+            ("test token", 0, "some_valid_authorization_code", True, True, False),
+            # invalid authorization code from callback
+            ("test token", 0, None, True, True, True),
+            # invalid state code from callback
+            ("test token", 0, None, False, True, True),
```

Comment:
This test method is getting a bit large. Maybe we need to think about factorizing the setup so that it can be split? Not sure what the best split would be, though :thinking:

Not a blocker to merge, just something to think about.

Reply:
I thought about it. The thing is there are so many things to mock that we must keep a reference to. I'll give it a try.